### PR TITLE
fix 0.x compatibility issue in functions setConfiguration() and setAltInterface()

### PR DESCRIPTION
**Pull request by AndreySV**
Updated patches, that we have discussed here: https://github.com/walac/pyusb/pull/103

### CODE REVIEW

File: usb/legacy.py [hunk 1]
```diff
@@ -256,7 +256,13 @@ def setConfiguration(self, configuration):
         Arguments:
             configuration: a configuration value or a Configuration object.
         """
-        self.dev.set_configuration(configuration)
+        if isinstance(configuration, Configuration):
+           conf_num = configuration.value
+        else:
+           conf_num = configuration
+
+        self.dev.set_configuration(conf_num)
```
Comment (walac) [inline usb/legacy.py]: hrm I think this is better:

```python
if isinstance(configuration, Configuration):
    configuration = configuration.value

self.dev.set_configuration(configuration)
```

File: usb/legacy.py [hunk 2]
```diff
@@ -264,7 +270,12 @@ def setAltInterface(self, alternate):
         Arguments:
             alternate: an alternate setting number or an Interface object.
         """
-        self.dev.set_interface_altsetting(self.__claimed_interface, alternate)
+        if isinstance(alternate, Interface):
+           alt_num = alternate.alternateSetting
+        else:
+           alt_num = alternate
+
+        self.dev.set_interface_altsetting(self.__claimed_interface, alt_num)
```
Comment (walac) [inline usb/legacy.py]: The same as for configuration applies here.

**Comment by walac** [comment]
Tip: you don't need to open a num PR, just make the fixes, squash de commits and do a force push that this PR will be updated.

**Comment by AndreySV** [comment]
I fixed patches according to your comments. Additionaly I fixed similar problem in detachKernelDriver and reworked a little bit claimInterface.

**Comment by walac** [comment]
Merged, thank you.

**Pull request merged**